racing/racing_test.go

```diff
@@ -0,0 +1,97 @@
+package main
+
+import (
+	"context"
+	"testing"
+	"time"
+
+	"github.com/google/go-cmp/cmp"
+	"github.com/stretchr/testify/assert"
+	"google.golang.org/genproto/googleapis/rpc/errdetails"
+	"google.golang.org/grpc/codes"
+	"google.golang.org/protobuf/testing/protocmp"
+
+	"git.neds.sh/matty/entain/racing/grpctest"
+	"git.neds.sh/matty/entain/racing/proto/racing"
+)
+
+func TestRacingListRaces(t *testing.T) {
+	if testing.Short() {
+		t.Skip("skipping test in short mode.")
+	}
+	t.Parallel()
+
+	for _, tc := range []struct {
+		name        string
+		giveContext context.Context
+		giveRequest *racing.ListRacesRequest
+		expect      *racing.ListRacesResponse
+		errAssert   func(*testing.T, error) bool
+	}{
+		{
+			name:        "success",
+			giveContext: context.Background(),
+			giveRequest: &racing.ListRacesRequest{
+				OrderBy: "number desc",
+				Filter: &racing.ListRacesRequestFilter{
+					MeetingIds:   []int64{9},
+					VisibileOnly: true,
+				},
+			},
+			// WARNING: This test relies on data seeded in the db package.
```

Comment:
For this type of integration test I'd usually advocate for spinning up the dependency for the server to connect to (e.g containerised DB).